src/main/java/org/myrobotlab/document/workflow/WorkflowWorker.java

```diff
@@ -63,6 +66,7 @@ public void run() {
           processing = true;
           // process from the start of the workflow
           processDocumentInternal(doc, 0);
+          pipeline.invoke("publishDocument", doc);

```

Comment:
If we don't want to maintain a reference to the pipeline and this is the only reason we need it, we could instead pass its name and then use Runtime.invoke() with its name to auto route to the pipeline, I think that works anyway, not at my desktop to check right now